### PR TITLE
{*}[foss/2021b] NCO v5.0.3, ESMF v8.2.0, netCDF-C++4 v4.3.1, ...

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
WIP because there are still some issues to resolve with NCO:
* [x] `libnco_c++.a` is missing (fixed)
* [x] 'Error 127: Command not found' relates to `ncks` (red herring in build output, verified with sanity check command)

### CODE REVIEW

File: easybuild/easyconfigs/n/NCO/NCO-5.0.3-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+dependencies = [
+    ('UDUNITS', '2.2.28'),
+    ('expat', '2.4.1'),
+    ('ANTLR', '2.7.7', '-Java-11'),
+    ('libdap', '3.20.8'),
+    ('GSL', '2.7'),
+    ('netCDF', '4.8.1'),
+    ('ESMF', '8.2.0'),  # ncremap needs ESMF_RegridWeightGen
+]
+
```
Comment (zarybnicky) [inline easybuild/easyconfigs/n/NCO/NCO-5.0.3-foss-2021b.eb:30]:
```suggestion
+ configopts = "--enable-nco_cplusplus"
+
```

File: easybuild/easyconfigs/n/NCO/NCO-5.0.3-foss-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+sanity_check_paths = {
+    'files': ['bin/nc%s' % x for x in ('ap2', 'atted', 'bo', 'diff', 'ea', 'ecat', 'es',
+                                       'flint', 'ks', 'pdq', 'ra', 'rcat', 'rename', 'wa')] +
+             ['lib/libnco.a', 'lib/libnco.%s' % SHLIB_EXT, 'lib/libnco_c++.a', 'lib/libnco_c++.%s' % SHLIB_EXT],
+    'dirs': ['include'],
+}
+
```
Comment (zarybnicky) [inline easybuild/easyconfigs/n/NCO/NCO-5.0.3-foss-2021b.eb:37]:
```suggestion
sanity_check_commands = [
    "ncks -O -7 --cnk_dmn time,10 "
    "%(builddir)s/%(namelower)s-%(version)s/data/in.nc %(builddir)s/%(namelower)s-%(version)s/data/in4.cdl"
]

```

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node2613.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/e686e0d2abcbebe25974f3c9c4ed4334 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3511.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/cdae420d405711bbd9f5878c16be1d2d for a full test report.

**Review by migueldiascosta — APPROVED** [review]
lgtm

**Comment by migueldiascosta** [comment]
Going in, thanks @boegel!

**Pull request merged**